facebook/react-native
# added the option to pass 'bundleCommand' to react.gradle

This will enable Android app to leverage the `unbundle` option of the packager. The default value will be `bundle` which is the default behaviour today.

## Motivation
We use react native in a production application and we noticed (by using production monitoring) that about 2 seconds of our app bootstrap time is from MainActivity.onResume() to the first line of the js bundle.
Our bundle size is about 4.5MB and most of it is not needed in order to load the first view of the application. After some investigation we found that using `unbundle` as our bundling strategy gives way better results (bundle load time was reduced from ~2 seconds to ~1 second)

## What existing problem does the pull request solve?
The standard way to create Android react native app is by `gradle assmebleRelease` that among the other tasks execute the `bundleReleaesJsAndAssets` task from `react.gradle`
This task in running the `bundle` command on the packager and does not expose an option to run `unbundle`. This PR adds a `bundleCommand` option (similar to other override options that available on `react.gradle`)

## Test Plan
This test scenario was performed manually to verify that everything is working as expected:
1. create new react application with `react-native init`
2. run `assembleRelease`
3. verify on `app/build/intermediates/assets/release` that `index.android.js` file exists
4. run the app and verify the main page is loading
5. go to `build.gradle` and add:
`project.ext.react = [
    bundleCommand: "unbundle"
]`
6. run `assembleRelease` again
7. verify on `app/build/intermediates/assets/release` that single index.android.js file exists and js-modules folder exists with multiple chunks
8. run the app and verify the main page is loading

By doing all these steps I managed to run the app in both scenarios
